Change private variables to protected

seifer8ff opened the pull request:
To allow extending the highcharts-chart component.

gregjacobs commented:
Why do you want to extend the Highcharts component exactly? I have a feeling that there might be a better option rather than turning to inheritance 

Can you use composition instead?

seifer8ff commented:
I'm trying to expose Highcharts chart events as Angular event Outputs. I wanted to avoid adding the events to the chart options configuration to avoid overwriting any existing events in the chart options passed to the component.

For instance, doing something like: 
`this.Highcharts.addEvent(this.chart.xAxis[0], 'afterSetExtremes', this.overrideAfterSetExtremes.bind(this));` where overrideAfterSetExtremes emits an event from an Angular Output. 

I did consider wrapping highcharts-chart component with another component and accessing the chart in the callback, but I thought it was cleaner to just extend it.

KacperMadej commented:
Hi @seifer8ff 

As I understand the case this is a new feature and the modification will allow easier / cleaner extending of the component? It would be easier to start an issue first, but let's continue in this PR.

We'll need to create some documentation to show how the feature could be used, so if you have any pointers, demo code, materials, etc. then please share.

There seems to be some reasons for the change. Are there any reasons against it? @gregjacobs  @seifer8ff 

* _Additional info about commit message:_
Currently used CHANGELOG.md requires commits as explained here - see [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines. - so they will end up in the changelog.

seifer8ff commented:
Sorry for the delay; I just returned from vacation. I fixed up my commits, and added some pared down example code below: 

Extend HighchartsChartComponent to add an Output on drawing the crosshair:
```javascript
import {ChangeDetectionStrategy, Component, ElementRef, EventEmitter, OnInit, Output} from '@angular/core';
import {HighchartsChartComponent} from 'highcharts-angular';
import {PointObject} from 'highcharts';

@Component({
  selector: 'app-highcharts-base',
  templateUrl: './highcharts-base.component.html',
  styleUrls: ['./highcharts-base.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class HighchartsBaseComponent extends HighchartsChartComponent implements OnInit {
  @Output() drawCrosshair: EventEmitter<PointObject> = new EventEmitter<PointObject>();

  constructor(el: ElementRef) {
    super(el);
  }

  ngOnInit() {
    // since this.chart is protected in the base component, it can be accessed in the inherited component
    this.Highcharts.addEvent(this.chart.xAxis[0], 'drawCrosshair', this.overrrideDrawCrosshair.bind(this));
  }

  private overrrideDrawCrosshair(e) {
    this.drawCrosshair.emit(e.point.x);
  }

}
```


Listen for newly added Output:
```html
<app-highcharts-base
      #chartContainer
      constructorType="chart"
      [Highcharts]="highcharts"
      [options]="chart_options"
      [callbackFunction]="afterChartBuild.bind(this)"
      [(update)]="update_flag"
      [oneToOne]="true"
      (drawCrosshair)="handleCrosshair($event)">
    </app-highcharts-base>
```

seifer8ff commented:
Closing this as the existing component can be wrapped to add the functionality I need.
